### PR TITLE
release 2.22

### PR DESCRIPTION
**Pull request by chessai**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
 # `chainweb-node` Changelog
 
+## 2.22 (2023-11-XX)
```
Comment (chessai) [inline CHANGELOG.md:3]: Which day are we actually doing the release? Today?

File: src/Chainweb/Version/Mainnet.hs [hunk 2]
```diff
@@ -141,7 +141,7 @@ mainnet = ChainwebVersion
         Chainweb219Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 3_774_423) -- 2023-06-02 00:00:00+00:00
         Chainweb220Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 4_056_499) -- 2023-09-08 00:00:00+00:00
         Chainweb221Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 4_177_889) -- 2023-10-20 00:00:00+00:00
-        Chainweb222Pact -> AllChains ForkNever -- TODO: fill in
+        Chainweb222Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 4_335_753) -- 2023-12-14 00:00:00+00:00
```
Comment (chessai) [inline src/Chainweb/Version/Mainnet.hs:144]: Should there be a `Chainweb223Pact` that returns `ForkNever`?

File: CHANGELOG.md (hunk 1 above)
Comment (edmundnoble) [reply CHANGELOG.md:3]: We can consider today to be the release cut date, yeah. Shipping is a week out.

File: node/ChainwebNode.hs [hunk 3]
```diff
@@ -526,10 +526,10 @@ pkgInfoScopes =
 -- -------------------------------------------------------------------------- --
 -- main
 
--- SERVICE DATE for version 2.21
+-- SERVICE DATE for version 2.22
 --
 serviceDate :: Maybe String
-serviceDate = Just "2023-12-13T00:00:00Z"
+serviceDate = Just "2024-03-06T00:00:00Z"
```
Comment (edmundnoble) [inline node/ChainwebNode.hs:532]: Hm this is Wednesday not Thursday. So is Dec 13. This is good with me though.

File: src/Chainweb/Version/Mainnet.hs (hunk 2 above)
Comment (edmundnoble) [reply src/Chainweb/Version/Mainnet.hs:144]: I actually really like the idea of handling that this release. Sure, add it in.

File: tools/calculate-release/CalculateRelease.hs [hunk 4]
```diff
@@ -48,8 +48,8 @@ main = do
 
     putStrLn $ "Mainnet fork height (at " <> show mainnetForkTime <> "): " <> show mainnetForkHeight
     putStrLn $ "Testnet fork height (at " <> show testnetForkTime <> "): " <> show testnetForkHeight
-    let nextServiceDateDay = addDays (8 * 7) serviceDateDay
-    putStrLn $ "Next service date (+8 weeks): " <> show nextServiceDateDay
+    let nextServiceDateDay = addDays (12 * 7) serviceDateDay
+    putStrLn $ "Next service date (+12 weeks): " <> show nextServiceDateDay
     where
```
Comment (edmundnoble) [inline tools/calculate-release/CalculateRelease.hs:53]: :+1:

Comment (edmundnoble) [reply tools/calculate-release/CalculateRelease.hs:53]: 3859808 - 3805981 (current testnet height) = 53827 blocks.
53827 blocks / ~2 blocks per minute = 26914 minutes.
26914 minutes / 60 = 449 hours.
449 hours / 24 hours in a day = 18 and a bit days.
It's 18 and a bit days until Dec 14.

File: src/Chainweb/Version/Testnet.hs [hunk 5]
```diff
@@ -121,7 +121,7 @@ testnet = ChainwebVersion
         Chainweb219Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_299_753 -- 2023-06-01 12:00:00+00:00
         Chainweb220Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_580_964 -- 2023-09-08 12:00:00+00:00
         Chainweb221Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_702_250 -- 2023-10-19 12:00:00+00:00
-        Chainweb222Pact -> AllChains ForkNever -- TODO: fill in
+        Chainweb222Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_859_808 -- 2023-12-13 12:00:00+00:00
```
Comment (edmundnoble) [inline src/Chainweb/Version/Testnet.hs:124]: 3859808 - 3805981 (current testnet height) = 53827 blocks.
53827 blocks / ~2 blocks per minute = 26914 minutes.
26914 minutes / 60 = 449 hours.
449 hours / 24 hours in a day = 18 and a bit days.
It's 18 and a bit days until Dec 14.

File: src/Chainweb/Version/Mainnet.hs (hunk 2 above)
Comment (edmundnoble) [reply src/Chainweb/Version/Mainnet.hs:144]: 4335753 - 4280488 (current mainnet height) = 55265.
Close to the testnet number below.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,28 @@
...
+ on **2023-12-13T:00:00Z**. Node administrators must upgrade to this version before
+ that date.
+ 
+ This version will expire on **2023-03-06T:00:00Z**.
+ 
+ To upgrade, pull the latest docker image or download the binary and restart the node.
+ 
+ Changes:
+ 
+Pact support for webauthn keys (#1779)
```
Comment (emilypi) [inline CHANGELOG.md:14]: I'd amend this to say "node support for webauthn signers, scoped signatures, and webauthn keyset formats in Pact"

File: node/ChainwebNode.hs (hunk 3 above)
Comment (emilypi) [reply node/ChainwebNode.hs:532]: I'm okay with that.

File: src/Chainweb/Version/Mainnet.hs (hunk 2 above)
Comment (emilypi) [reply src/Chainweb/Version/Mainnet.hs:144]: Works for me

File: src/Chainweb/Version/Testnet.hs (hunk 5 above)
Comment (chessai) [reply src/Chainweb/Version/Testnet.hs:124]: so that's correct, yes?

**Review by edmundnoble — APPROVED** [review]

**Review by thoughtpolice — APPROVED** [review]
No comments on the height changes, but otherwise straightforward and LGTM.

**Pull request merged**